plugins/lua/zone.lua

```diff
@@ -0,0 +1,542 @@
+local _ENV = mkmodule('plugins.zone')
+local gui = require('gui')
+local overlay = require('plugins.overlay')
+local widgets = require('gui.widgets')
+
+ZoneOverlay = defclass(ZoneOverlay, overlay.OverlayWidget)
+ZoneOverlay.ATTRS{
+    default_pos={x=46,y=39},
+    default_enabled=true,
+    viewscreens='dwarfmode',
+    frame={w=42, h=15},
+    frame_style=gui.MEDIUM_FRAME,
+    frame_background=gui.CLEAR_PEN,
+}
+
+local function zone_assign_is_active()
+    return dfhack.gui.matchFocusString('dwarfmode/UnitSelector') and (df.global.game.main_interface.unit_selector.context == 0 or df.global.game.main_interface.unit_selector.context == 1)
```

Comment:
The focus string part should move into the viewscreen attribute on the overlay. See `orders.lua` for an example

Reply:
And should the context value be part of the focus string as well?